iceoryx2-ffi/cxx/include/iox2/port_factory_publisher.hpp

```diff
@@ -16,28 +16,85 @@
 #include "iox/expected.hpp"
-#include "publisher.hpp"
-#include "service_type.hpp"
+#include "iox2/internal/iceoryx2.hpp"
+#include "iox2/publisher.hpp"
+#include "iox2/service_type.hpp"
 
 #include <cstdint>
 
 namespace iox2 {
+/// Defines the strategy the [`Publisher`] shall pursue in
+/// [`send_sample(`] or
+/// [`Publisher::send_copy()`] when the buffer of a
+/// [`Subscriber`] is full and the service does not overflow.
 enum class UnableToDeliverStrategy : uint8_t {
+    /// Blocks until the [`Subscriber`] has consumed the
+    /// [`Sample`] from the buffer and there is space again
     Block,
+    /// Do not deliver the [`Sample`].
     DiscardSample
 };
 
+/// Factory to create a new [`Publisher`] port/endpoint for
+/// [`MessagingPattern::PublishSubscribe`] based communication.
 template <ServiceType S, typename Payload, typename UserHeader>
 class PortFactoryPublisher {
+    /// Sets the [`UnableToDeliverStrategy`].
     IOX_BUILDER_OPTIONAL(UnableToDeliverStrategy, unable_to_deliver_strategy);
+
+    /// Defines how many [`SampleMut`] the [`Publisher`] can loan with
+    /// [`Publisher::loan()`] or [`Publisher::loan_uninit()`] in parallel.
     IOX_BUILDER_OPTIONAL(uint64_t, max_loaned_samples);
+
+    /// Sets the maximum slice length that a user can allocate with
+    /// [`Publisher::loan_slice()`] or [`Publisher::loan_slice_uninit()`].
     IOX_BUILDER_OPTIONAL(uint64_t, max_slice_len);
 
   public:
-    auto create() && -> iox::expected<Publisher<S, Payload, UserHeader>, PublisherCreateError> {
-        IOX_TODO();
-    }
+    PortFactoryPublisher(const PortFactoryPublisher&) = delete;
+    PortFactoryPublisher(PortFactoryPublisher&&) = default;
+    auto operator=(const PortFactoryPublisher&) -> PortFactoryPublisher& = delete;
+    auto operator=(PortFactoryPublisher&&) -> PortFactoryPublisher& = default;
+    ~PortFactoryPublisher() = default;
+
+    /// Creates a new [`Publisher`] or returns a [`PublisherCreateError`] on failure.
+    auto create() && -> iox::expected<Publisher<S, Payload, UserHeader>, PublisherCreateError>;
+
+  private:
+    template <ServiceType, typename, typename>
+    friend class PortFactoryPublishSubscribe;
+
+    explicit PortFactoryPublisher(iox2_port_factory_publisher_builder_h handle);
+
+    iox2_port_factory_publisher_builder_h m_handle;
 };
+
+template <ServiceType S, typename Payload, typename UserHeader>
+inline PortFactoryPublisher<S, Payload, UserHeader>::PortFactoryPublisher(iox2_port_factory_publisher_builder_h handle)
+    : m_handle { handle } {
+}
+
+template <ServiceType S, typename Payload, typename UserHeader>
+inline auto
+PortFactoryPublisher<S, Payload, UserHeader>::create() && -> iox::expected<Publisher<S, Payload, UserHeader>,
+                                                                           PublisherCreateError> {
+    auto* ref_handle = iox2_cast_port_factory_publisher_builder_ref_h(m_handle);
+
+    m_unable_to_deliver_strategy.and_then([](auto) { IOX_TODO(); });
```

Comment:
Of all the available options, I think this is one which we need to support with the initial bindings.

Reply:
Done, see #325